### PR TITLE
feat: Implement float*interval and interval/float

### PR DESCRIPTION
**Pull request by mcheshkov**

### CODE REVIEW

**Review by srh — APPROVED** [review]
I mostly looked at the arithmetic and how it compares to postgres logic.

File: datafusion/core/tests/sql/expr.rs [hunk 1]
```diff
@@ -725,6 +725,164 @@ async fn test_interval_expressions() -> Result<()> {
...
+        "2.0",
+        "interval '1 second'",
+        "0 years 0 mons 0 days 0 hours 0 mins 2.000000000 secs"
+    );
+    // This test prints as -1.-500000000 secs, that looks like a bug in printing
+    // TODO fix it
+    // test_mul!(
+    //     "-1.5",
+    //     "interval '1 second'",
+    //     "0 years 0 mons 0 days 0 hours 0 mins -1.500000000 secs"
```
Comment (srh) [inline datafusion/core/tests/sql/expr.rs:822]: I thought I fixed this.

Comment (srh) [reply datafusion/core/tests/sql/expr.rs:822]: That was fixed in cube-js/cube; maybe it copied the logic from arrow-datafusion and the fix is similar.

https://github.com/cube-js/cube/pull/8358/files

**Review by paveltiunov — APPROVED** [review]

File: datafusion/core/tests/sql/expr.rs (hunk 1 above)
Comment (srh) [reply datafusion/core/tests/sql/expr.rs:822]: This behavior is actually in the arrow-rs dependency, in its make_string_interval_day_time macro.

**Pull request merged**